support/sier-codec/src/json.rs

```diff
@@ -0,0 +1,63 @@
+use crate::{Error, Object, StructDef, Type, Value};
+use serde_json::Value as SerdeValue;
+use std::sync::Arc;
+
+pub fn parse_serde_value<'a>(value: &SerdeValue, type_: &'a Type) -> Result<Value<'a>, Error<'a>> {
+    match value {
+        SerdeValue::Null => unimplemented!(),
+        SerdeValue::Bool(_bool) => unimplemented!(),
+        SerdeValue::Number(number) => {
+            if *type_ != Type::U64 {
+                return Err(Error::InvalidJson);
+            }
+
+            let n = number.as_u64().ok_or_else(|| unimplemented!())?;
+            Ok(Value::U64(n))
+        }
+        SerdeValue::String(s) => {
+            if *type_ != Type::String {
+                return Err(Error::InvalidJson);
+            }
+
+            Ok(Value::String(s.to_string()))
+        }
+        SerdeValue::Array(vec) => {
+            if let Type::List(arr_type) = type_ {
+                let mut list = vec![];
+                for arr_value in vec {
+                    let val = parse_serde_value(arr_value, arr_type)?;
+                    list.push(val);
+                }
+                Ok(Value::List(list))
```

Comment:
You can collect an `Iterator<Result<T, E>>` into a `Result<Vec<T>, E>`. So might be nicer as:

```rs
let list = vec
    .into_iter()
    .map(|val| parse_serde_value(val, arr_type))
    .collect::<Result<_, _>>()?;
Ok(Value::List(list))
```